app/controllers/setting_controller.rb

```diff
@@ -52,10 +54,10 @@ def create
     render(nothing: true, status: 405) && return unless request.content_type == 'application/json'
     render(nothing: true, status: 405) && return unless params.key?(:marketplace_url)
 
+    stop_threads
```

Comment:
Why that?

Reply:
If an already running consumer receives another start request, the worker threads are restarted (i.e. old threads are stopped, new thread are started).